src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilterElement.java

```diff
@@ -253,11 +223,28 @@ public boolean equals(Object other) {
             return false;
         }
         final XpathFilterElement xpathFilter = (XpathFilterElement) other;
-        return Objects.equals(filePattern, xpathFilter.filePattern)
-                && Objects.equals(checkPattern, xpathFilter.checkPattern)
-                && Objects.equals(messagePattern, xpathFilter.messagePattern)
+        return Objects.equals(getPatternSafely(fileRegexp),
+                getPatternSafely(xpathFilter.fileRegexp))
+                && Objects.equals(getPatternSafely(checkRegexp),
+                getPatternSafely(xpathFilter.checkRegexp))
+                && Objects.equals(getPatternSafely(messageRegexp),
+                getPatternSafely(xpathFilter.messageRegexp))
```

Comment:
lines starting with `getPatternSafely` should indented +4 over the other lines.

Reply:
done